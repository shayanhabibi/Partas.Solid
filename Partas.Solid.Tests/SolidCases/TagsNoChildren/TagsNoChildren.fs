module Partas.Solid.Tests.SolidCases.TagsNoChildren

open Partas.Solid
open Fable.Core

type [<Erase>] MyTag() =
    inherit RegularNode()
    [<SolidTypeComponent>]
    member props.typeDef =
        div()

type [<Erase>] TagWithProp() =
    inherit RegularNode()
    [<SolidTypeComponent>]
    member props.typeDef =
        div(class' = "SomeClass")

type [<Erase>] TagWithProps() =
    inherit RegularNode()
    [<SolidTypeComponent>]
    member props.typeDef =
        div(class' = "SomeClass", draggable = "drag")


