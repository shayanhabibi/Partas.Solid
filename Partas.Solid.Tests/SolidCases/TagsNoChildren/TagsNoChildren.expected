import { splitProps } from "solid-js";
export function MyTag(props) {
    const [local, others] = splitProps(props, []);
    return <div />;
}
export function TagWithProp(props) {
    const [local, others] = splitProps(props, []);
    return <div class="SomeClass" />;
}
export function TagWithProps(props) {
    const [local, others] = splitProps(props, []);
    return <div class="SomeClass"
        draggable="drag" />;
}
